File: frontend/src/api/chat_import.ts
import { fetchApi } from './client';

export interface ImportSummary {
  total_conversations: number;
  total_topics: number;
  new_conversations: number;
  updated_conversations: number;
  skipped_conversations: number;
  domains: Record<string, number>;
}

export interface ConversationPreview {
  external_id: string;
  title: string;
  status: string; // "new", "updated", "skipped"
  domain: string;
  topics_count: number;
  messages_count: number;
}

export interface ImportPreviewResponse {
  job_id: string;
  summary: ImportSummary;
  conversations_preview: ConversationPreview[];
}

export const chatImportApi = {
  startImport: async (file: File, provider: string): Promise<{ job_id: string; status: string }> => {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('provider', provider);

    return fetchApi('/connectors/chats/import', {
      method: 'POST',
      body: formData,
    });
  },

  getStatus: async (jobId: string): Promise<{ job_id: string; status: string; error_message?: string }> => {
    return fetchApi(`/connectors/chats/import/${jobId}`);
  },

  getPreview: async (jobId: string): Promise<ImportPreviewResponse> => {
    return fetchApi(`/connectors/chats/import/${jobId}/preview`);
  },

  commitImport: async (jobId: string, mode: string, selectedIds?: string[]): Promise<{ status: string }> => {
    return fetchApi(`/connectors/chats/import/${jobId}/commit`, {
      method: 'POST',
      body: JSON.stringify({
        mode,
        selected_external_ids: selectedIds
      })
    });
  }
};

File: backend/app/schemas/chat_import.py
from pydantic import BaseModel
from typing import Dict, List, Optional
from datetime import datetime
import uuid


class ImportJobResponse(BaseModel):
    job_id: uuid.UUID
    status: str
    created_at: datetime


class ImportSummary(BaseModel):
    total_conversations: int
    total_topics: int
    new_conversations: int
    updated_conversations: int
    skipped_conversations: int
    domains: Dict[str, int]


class ConversationPreview(BaseModel):
    external_id: str
    title: str
    status: str  # "new", "updated", "skipped"
    domain: str
    topics_count: int
    messages_count: int


class ImportPreviewResponse(BaseModel):
    job_id: uuid.UUID
    summary: ImportSummary
    conversations_preview: List[ConversationPreview]


class CommitImportRequest(BaseModel):
    mode: str  # "FULL", "NEW_ONLY", "UPDATE", "SELECTIVE"
    selected_external_ids: Optional[List[str]] = None
